astropy/io/fits/tests/test_convenience.py

```diff
@@ -61,3 +60,36 @@ def test_table_to_hdu(self, tmpdir):
         assert isinstance(hdu, fits.BinTableHDU)
         filename = str(tmpdir.join('test_table_to_hdu.fits'))
         hdu.writeto(filename, overwrite=True)
+
+    def test_printdiff(self):
+        """
+        Test that FITSDiff can run the different inputs without crashing.
+        """
+
+        # Testing different string input options
+        assert printdiff(self.data('arange.fits'), self.data('blank.fits')) is None
+        assert printdiff(self.data('arange.fits'), self.data('blank.fits'), ext=0) is None
+        assert printdiff(self.data('o4sp040b0_raw.fits'), self.data('o4sp040b0_raw.fits'), extname='sci') is None
+
+        # Test HDU object inputs
+        with fits.open(self.data('stddata.fits'), mode='readonly') as in1:
+            with fits.open(self.data('checksum.fits'), mode='readonly') as in2:
+
+                assert printdiff(in1[0], in2[0]) is None
+
+                try:
```

Comment:
If this is *expected* to raise error, here is a better way to do this:
```python
with pytest.raises(ValueError) as e:  # No need for "as e" if not asserting below
    printdiff(in1[0], in2[0], ext=0)
    assert "expected message" in str(e)  # This part is optional
```

Sorry, I know we spoke offline about this. `pytest.raises` slipped my mind at that time.